core/shared/src/main/scala/org/typelevel/log4cats/KernelLogLevel.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2018 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.log4cats
+
+import cats.Order
+
+final case class KernelLogLevel(name: String, value: Int) {
+  def namePadded: String = KernelLogLevel.padded(this)
+
+  KernelLogLevel.add(this)
+}
+
+object KernelLogLevel {
+  private var maxLength = 0
+
+  private var map = Map.empty[String, KernelLogLevel]
+  private var padded = Map.empty[KernelLogLevel, String]
+
+  implicit final val orderKernelLogLevel: Order[KernelLogLevel] =
+    Order.by[KernelLogLevel, Int](-_.value)
+
+  // For Java/legacy interop, if needed (not implicit)
+  val LevelOrdering: Ordering[KernelLogLevel] =
+    Ordering.by[KernelLogLevel, Int](_.value).reverse
```

Comment:
Java interop is a non-issue, since attempting to use cats-effect from Java would be painful (at best) and the idioms are all sorts of incompatible.

There's an implicit which provides an `Ordering[A]` if there is an `Order[A]` in scope, so this can be removed.